File: Models/Models/CompanyCSE.cs
using eMaestroD.Models.Custom;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eMaestroD.Models.Models
{
    public class CompanyCSE : IEntityBase
    {
        [Key]
        [HiddenOnRender]
        public int CSEID { get; set; }
        [HiddenOnRender]
        public int? CompID { get; set; }
        [HiddenOnRender]
        public int? vendID { get; set; }
        [NotMapped]
        [DisplayName(Name ="Supplier Name")]
        public string? vendName { get; set; }
        [DisplayName(Name = "CSE Name")]
        public string? RepName { get; set; }
        [DisplayName(Name = "Address")]
        public string? Address1 { get; set; }
        [DisplayName(Name = "Email")]
        public string? email { get; set; }
        [DisplayName(Name = "Phone")]
        public string? Cell { get; set; }
        [HiddenOnRender]
        public bool? Active { get; set; }
        [HiddenOnRender]
        public bool? IsDefault { get; set; }
        [NotMapped]
        [HiddenOnRender]
        public List<CSECustomer>? CSECustomer { get; set; }
        //[NotMapped]
        //[HiddenOnRender]
        //public List<MseMapArea> MseMapArea { get; set; }
    }
}

File: eMaestroD.NgUI/src/app/Manage/Models/company-cse.ts
import { CSECustomer } from "./csecustomer";
import { MseMapArea } from "./mse-map-area";

export class CompanyCSE {
  CSEID?: number | null = null; // Nullable number for CSEID
  CompID?: number | null = null; // Nullable number for CompID
  vendID?: number | null = null; // Nullable number for vendID
  vendName?: string | null = null; // Nullable string for vendName
  RepName?: string | null = null; // Nullable string for RepName
  Address1?: string | null = null; // Nullable string for Address1
  email?: string | null = null; // Nullable string for email
  Cell?: string | null = null; // Nullable string for Cell
  Active?: boolean | null = null; // Nullable boolean for Active
  IsDefault?: boolean | null = null; // Nullable boolean for IsDefault
  CSECustomer?: CSECustomer[] = []; // List of CSECustomer objects (empty array by default)
}
